fedmsg_meta_fedora_infrastructure/fasshim.py

```diff
@@ -90,6 +90,49 @@ def avatar_url_from_email(email, size=64, default='retro', dns=False):
         hash = md5(email.encode('utf-8')).hexdigest()
         return "https://seccdn.libravatar.org/avatar/%s?%s" % (hash, query)
 
+def make_fasjson_cache(**config):
+    global _fas_cache
+    if _fas_cache:
+        return _fas_cache
+    
+    log.warn("No previous fasjson cache found. Looking to rebuild.")
+
+    try:
+        import fasjson_client
+    except ImportError:
+        log.warn("No fasjson-client installed.  Not caching fasjson.")
+        return {}
+    
+    creds = config['fas_credentials']
+    
+    default_url = 'https://admin.fedoraproject.org/accounts/'
+    
+    try:
+        client = fasjson_client.Client(url=creds.get('base_url', default_url))
+    except fasjson_client.errors.ClientSetupError as e:
+        log.error("Something went wrong setting up fasjson client with error: %s" % e)
+        return {}
+    try:
+        _add_to_cache(list(client.list_all_entities("users")))
+    except fasjson_client.errors.APIError as e:
+        log.error("Something went wrong building cache with error: %s" % e)
+        continue
```

Comment:
When I run the tests with tox I get `SyntaxError: 'continue' not properly in loop`. It should be `return {}`.

Reply:
thanks, not sure how that slipped through. fixed now.